lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -0,0 +1,119 @@
+"""
+Test for course API
+"""
+
+from django.http import Http404
+from django.test import RequestFactory
+from lettuce import world
+
+from opaque_keys import InvalidKeyError
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import ToyCourseFactory
+
+from lms.djangoapps.course_api.api import course_detail, list_courses
+
+
+class CourseApiTestMixin(object):
+
+    expected_course_data = {
+        'blocks_url': '/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall',
+        'course_image': u'/c4x/edX/toy/asset/just_a_test.jpg',
+        'description': u'A course about toys.',
+        'end': None,
+        'enrollment_end': None,
+        'enrollment_start': None,
+        'id': u'edX/toy/2012_Fall',
+        'name': u'Toy Course',
+        'number': u'toy',
+        'org': u'edX',
+        'start': u'2015-07-17T12:00:00Z',
+        'start_display': u'July 17, 2015',
+        'start_type': u'timestamp',
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseApiTestMixin, cls).setUpClass()
+        cls.request_factory = RequestFactory()
+
+    @staticmethod
+    def create_course():
+        """
+        Create a course for use in test cases
+        """
+        return ToyCourseFactory.create()
+
+    @staticmethod
+    def create_user(username, email, password, is_staff):
+        """
+        Create a user as identified by username, email, password and is_staff.
+        """
+        user = world.UserFactory(
+            username=username,
+            email=email,
+            password=password,
+            is_staff=is_staff)
+        return user
+
+
+class TestGetCourseDetail(CourseApiTestMixin, ModuleStoreTestCase):
```

Comment:
Need to add test cases for authorization checks (`see_exists` permission) and authentication checks (AnonymousUser and AuthenticatedUser access).